data_structures/src/chain.rs

```diff
@@ -17,6 +7,18 @@ use std::{
     ops::{AddAssign, SubAssign},
     str::FromStr,
 };
+
+use bech32::{FromBase32, ToBase32};
+use bls_signatures_rs::{bn256, bn256::Bn256, MultiSignature};
+use failure::Fail;
+use ordered_float::OrderedFloat;
+use secp256k1::{
+    PublicKey as Secp256k1_PublicKey, SecretKey as Secp256k1_SecretKey,
+    Signature as Secp256k1_Signature,
+};
+use serde::{Deserialize, Serialize};
+
+use partial_struct::PartialStruct;
```

Comment:
Everything is better if alphabetically ordered :rofl: 
```suggestion
use bech32::{FromBase32, ToBase32};
use bls_signatures_rs::{bn256, bn256::Bn256, MultiSignature};
use failure::Fail;
use ordered_float::OrderedFloat;
use partial_struct::PartialStruct;
use secp256k1::{
    PublicKey as Secp256k1_PublicKey, SecretKey as Secp256k1_SecretKey,
    Signature as Secp256k1_Signature,
};
use serde::{Deserialize, Serialize};
```